web-api/src/v1/getCaseLambda.test.js

```diff
@@ -96,16 +96,15 @@ describe('getCaseLambda (which fails if version increase is needed, DO NOT CHANG
 
     expect(response.statusCode).toBe('200');
     expect(response.headers['Content-Type']).toBe('application/json');
-    expect(JSON.parse(response.body)).toHaveProperty(
-      'caseCaption',
-      expect.any(String),
-    );
-    expect(JSON.parse(response.body).assignedJudge).toBeUndefined();
-    expect(JSON.parse(response.body).contactPrimary).toBeUndefined();
-    expect(JSON.parse(response.body).noticeOfTrialDate).toBeUndefined();
-    expect(JSON.parse(response.body).status).toBeUndefined();
-    expect(JSON.parse(response.body).trialLocation).toBeUndefined();
-    expect(JSON.parse(response.body).userId).toBeUndefined();
+    const parsedResponse = JSON.parse(response.body);
+    expect(parsedResponse).toHaveProperty('caseCaption', expect.any(String));
+    expect(parsedResponse.assignedJudge).toBeUndefined();
+    expect(parsedResponse.noticeOfTrialDate).toBeUndefined();
+    expect(parsedResponse.trialLocation).toBeUndefined();
+    expect(parsedResponse.userId).toBeUndefined();
+    // contact and status is unavailable for user unassociated with the case
+    expect(parsedResponse.contactPrimary).toBeDefined();
+    expect(parsedResponse.status).toBeDefined();
```

Comment:
these should be undefined as per the earlier tests.

Reply:
As per discussion this morning, the `status` here was unintentionally missing from this API endpoint. It is included in the API contract and was set on some cases, but not always. This change is because for this specific test case, the case status _should have been_ and now _will_ be included. 

@kkoskelin, can you explain why `contactPrimary` is included as well? We talked about it this morning, and you mentioned that it was also not set sometimes when it should have been — but I can’t follow why it would have been missing before and is now set. Is there some filter which checks on `status`? 

In any case, the comment here is incorrect and contradicts the test assertions — probably worth dropping the comment so it doesn’t become outdated again.

Reply:
@adunkman The particular case used in this test was _sealed_, and previously had no `status` value.  It means that the marshalling function attempted to obtain the `contactPrimary` as seen here:
https://github.com/ustaxcourt/ef-cms/pull/1412/files#diff-384b0330bd6db4b05ec5bf32f098eb0c01feab98b385ef5653e6dbc59faf58f8R1603

However, the "else" block executed wasn't using the correct method of obtaining the contact info because this case was *not* actually New -- it was Calendared.  The contact information was most definitely available on the PublicCase entity instance, but our code was doing a poor job of attempting to retrieve it, and thus it previously had been undefined.